apps/studio/src/pages/sites/[siteId]/settings.tsx

```diff
@@ -1,8 +1,222 @@
+        notification: notificationEnabled ? notification : "",
+      })
+    },
+  )
+
+  return (
+    <>
+      <UnsavedSettingModal
+        isOpen={isOpen}
+        onClose={onClose}
+        nextUrl={nextUrl}
+      />
+      <form onSubmit={onClickUpdate}>
+        <Center pt="5.5rem">
+          <VStack w="48rem" alignItems="flex-start" spacing="1.5rem">
+            <FormControl isInvalid={!!errors.notification}>
+              <Text w="full" textStyle="h3-semibold">
+                Manage site settings
+              </Text>
+              <Infobox
+                textStyle="body-2"
+                textColor="base.content.strong"
+                size="sm"
+              >
+                Isomer Next is currently in Beta. To manage site settings that
+                are not displayed here, contact Isomer Support.
+              </Infobox>
+              <VStack
+                w="full"
+                alignItems="flex-start"
+                mt="3rem"
+                spacing="0.75rem"
+              >
+                <Text textStyle="h4">General</Text>
+                <HStack w="full" justifyContent="space-between" pt="0.5rem">
+                  <Text textColor="base.content.strong" textStyle="subhead-1">
+                    Display Site Notifications
+                  </Text>
+
+                  <Toggle
+                    w="full"
+                    label=""
+                    {...register("notificationEnabled")}
+                  />
+                </HStack>
+                <Text textColor="base.content.medium" textStyle="body-2">
+                  The site notification will always be visible on the site until
+                  it is dismissed by the user. <br />
+                  Use a notification to inform users of key alerts. Do not use a
+                  notification to advertise an event or a promotion.
+                </Text>
+                {notificationEnabled && (
+                  <>
+                    <Text
+                      textColor="base.content.strong"
+                      textStyle="subhead-1"
+                      pt="0.5rem"
+                    >
+                      Notification Text
+                    </Text>
+
+                    <Input
+                      isDisabled={!notificationEnabled}
+                      placeholder="Notification should be succinct and clear"
+                      maxLength={100}
+                      value={notification}
+                      {...register("notification", {
+                        required: true,
+                        minLength: {
+                          value: 1,
+                          message: "Notification must not be empty",
+                        },
+                        maxLength: {
+                          value: 100,
+                          message:
+                            "Notification must be 100 characters or less",
+                        },
+                      })}
```

Comment:
since you are using a zod schema, the validation should be on the schema itself. You should update the schema instead of declaring them here:

```ts
const setNotificationSchema = z.object({
  // ...
  notification.max(100, { message: "Notification must be 100 characters or less" })
```

and the validation errors should appear without any additional config in react hook form:
```ts
{...register("notification")}
```

Reply:
i realise there is some drift between what the client and backend schema expects. thinking about it more, why not just have them use the same schema? Server can also take in the `notificationsEnabled` boolean and handle accordingly. This may simplify a bunch of things.

Reply:
Okay, I think I will prefer to standardize the backend and client schema. This will also eliminate the need for a clearnotification endpoint(since we explicit pass notificationsEnabled: false to clear the notification.)